plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/thrift/TokenFetchingMetastoreClientFactory.java

```diff
@@ -106,11 +120,21 @@ private DelegationToken getDelegationToken(String username)
 
     private DelegationToken loadDelegationToken(String username)
     {
-        try (ThriftMetastoreClient client = createMetastoreClient()) {
-            return new DelegationToken(System.nanoTime(), client.getDelegationToken(username));
+        try {
+            // added retry and stats for the thrift delegation token
+            return (DelegationToken) Failsafe.with(retryPolicy).get((CheckedSupplier<DelegationToken>) () ->
+                            stats.getThriftDelegationToken().wrap(() -> {
+                                try (ThriftMetastoreClient client = createMetastoreClient()) {
+                                    return new DelegationToken(System.nanoTime(), client.getDelegationToken(username));
+                                }
+                            }
+                ).call());
         }
-        catch (TException e) {
```

Comment:
why not just
```
catch (TException e) {
  throw new TrinoException(HIVE_METASTORE_ERROR, e);
}
```
?

Reply:
was, not sure, if from the service, `TException` is wrapped in another exception type. Otherwise it should be like how you said.

Let me change it, it should be ok.

Reply:
<img width="970" alt="image" src="https://github.com/user-attachments/assets/7ccbf43a-0762-4480-816d-45907c7b2a47">

`Failsafe.with(retryPolicy).get((CheckedSupplier<DelegationToken>) () ->` 

does not throws `TException` therefore we can't catch it directly.

Reply:
You can do:
```
            catch (TException e) {
                throw new TrinoException(HIVE_METASTORE_ERROR, e);
            }
            catch (Exception e) {
                throw propagate(e);
            }
```